src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -1388,6 +1409,15 @@ void SelectQueryExpressionAnalyzer::appendWindowFunctionsArguments(
     }
 }
 
+void SelectQueryExpressionAnalyzer::appendExpressionsAfterWindowFunctions(ExpressionActionsChain & chain, bool /* only_types */)
+{
+    ExpressionActionsChain::Step & step = chain.lastStep(columns_after_window);
+    for (const auto & expression : syntax->expressions_with_window_function)
+    {
+        getRootActionsForWindowFunctions(expression->clone(), true, step.actions());
```

Comment:
So, generally, most of the work is done here. Have I got right that it's the part that drives the calculation of expressions for window functions?

Reply:
Yes, here we create actions for expression evaluation.